apis/core/lib/domain/cube-projects/import.ttl

```diff
@@ -0,0 +1,18 @@
+prefix cc: <https://cube-creator.zazuko.com/vocab#>
+PREFIX dcterms: <http://purl.org/dc/terms/>
+PREFIX dash: <http://datashapes.org/dash#>
+PREFIX schema: <http://schema.org/>
+PREFIX sh: <http://www.w3.org/ns/shacl#>
+
+[
+  a sh:NodeShape ;
+  sh:targetClass cc:CubeProject ;
+  sh:property
+    [
+      sh:path ( <https://cube-creator.zazuko.com/vocab#dataset> dcterms:identifier ) ;
+      sh:maxCount 0 ;
+      dash:hidden true ;
+      sh:message "Project metadata must not have a dcterms:identifier" ;
+    ]
+]
+.
```

Comment:
Most other validation loads shapes from the store albeit expecting a more resource-oriented flow. For project import, I added a dedicated handler middleware to validate the import.

I'm not sure, maybe it should actually also load shapes from the store? Right now seems that it might be start of some duplication

Reply:
I'm not sure either. 

What are the pros/cons of having the shape in the store?

I would inspect what the scope of this specific shape is. 

If that shape is used exclusively in the cube-creator application (part of its model) and is not used to publish public data, I would definitively keep the shape (in the form of a file or just code) in the code-creator codebase and not in the store. I think that diminishes the overall complexity.

If, on the other hand, the project entities are to be used by other applications, then the model (shape) should be shared between the apps, I think.

Reply:
The application itself can publish it's shapes if required.

Reply:
All right, thanks for challenging me on this.

The problem with some shapes is that the do not have a `sh:target*` property. Also, they are loaded on a per-request basis, based on `hydra:expects` of the current operation and not based on the data model. Slightly limiting, but not something I would like to touch now...

To reduce the duplication thought, it is definitely possible to re-use what shapes we have. I would not blindly load all shapes but for now only import the one for cube metadata resource. Might add more in the future as long as they have target properties.